### ISSUE TITLE
Plugin server doesn't URL-encode parameters when building database connection URL

### ISSUE THREAD

**Issue by joetrollo** [PostHog/posthog#13103]
## Bug description

Upon updating PostHog to v1.41.4 and v1.42.0, the new plugin server does not start and repeatedly crash-loops with the error `getaddrinfo ENOTFOUND posthog`. Unfortunately this error doesn't come with a stack trace, so I cannot identify what lines attempted to resolve `posthog`, and my attempts at searching the code base have been unsuccessful as well.

## How to reproduce

Unknown—this issue did not occur when running version 1.40.0 and only presented itself upon upgrading to >=1.41.4.

## Environment

- [ ] PostHog Cloud
- [X] self-hosted PostHog, version/commit: v1.41.4, v1.42.0, v1.43.0; Helm chart 30.0.0

## Additional context

Logs from the crash-looping pod:
```
▶️ Starting plugin server...
{"level":"info","time":1670009189771,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] ⚡ @posthog/plugin-server v1.10.5"}
{"level":"info","time":1670009189780,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] ℹ️ 4 workers, 10 tasks per worker"}
{"level":"info","time":1670009189781,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] ℹ️ Connecting to all services:"}
{"level":"info","time":1670009189809,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 🤔 Connecting to ClickHouse..."}
{"level":"info","time":1670009189875,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 👍 ClickHouse ready"}
{"level":"info","time":1670009189875,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 🤔 Connecting to Kafka..."}
{"level":"info","time":1670009189957,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 👍 Kafka ready"}
{"level":"info","time":1670009189957,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 🤔 Connecting to Postgresql..."}
{"level":"info","time":1670009189958,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 👍 Postgresql ready"}
{"level":"info","time":1670009189958,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 🤔 Connecting to Redis..."}
{"level":"info","time":1670009189965,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 👍 Redis ready"}
{"level":"info","time":1670009189965,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 🤔 Connecting to object storage..."}
{"level":"info","time":1670009189965,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 👍 Object storage ready"}
{"level":"info","time":1670009189983,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] ✅ Connected to Redis!"}
{"level":"error","time":1670009190029,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","error":"Error: getaddrinfo ENOTFOUND posthog\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:111:26)","msg":"[MAIN] 💥 Launchpad failure!"}
{"level":"info","time":1670009190030,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 💤  Shutting down gracefully..."}
{"level":"info","time":1670009190030,"pid":22,"hostname":"posthog-plugins-5496f989cf-l8hjw","msg":"[MAIN] 👋 Over and out!"}
```

JSON payload of Sentry event:
```
{
	"event_id": "eb8c5c56880f4c5796be089ee0c1d2f6",
	"project": [REDACTED],
	"release": null,
	"dist": null,
	"platform": "node",
	"message": "",
	"datetime": "2022-12-02T19:03:18.393000Z",
	"tags": [
		["environment", "production"],
		["handled", "yes"],
		["level", "error"],
		["mechanism", "generic"],
		["runtime", "node v18.9.1"],
		["runtime.name", "node"],
		["server_name", "posthog-plugins-794d9fc977-xvt94"]
	],
	"_metrics": {
		"bytes.ingested.event": 1259,
		"bytes.stored.event": 2450,
		"sample_rates": [{}]
	},
	"breadcrumbs": {
		"values": [{
			"timestamp": 1670007798.07,
			"type": "http",
			"category": "http",
			"level": "info",
			"data": {
				"method": "POST",
				"status_code": 200,
				"url": "https://[REDACTED]/?database=posthog&output_format_json_quote_64bit_integers=false"
			}
		}]
	},
	"contexts": {
		"runtime": {
			"name": "node",
			"version": "v18.9.1",
			"type": "runtime"
		}
	},
	"culprit": "GetAddrInfoReqWrap.onlookup [as oncomplete](node:dns)",
	"environment": "production",
	"exception": {
		"values": [{
			"type": "Error",
			"value": "getaddrinfo ENOTFOUND posthog",
			"stacktrace": {
				"frames": [{
					"function": "GetAddrInfoReqWrap.onlookup [as oncomplete]",
					"module": "node:dns",
					"filename": "node:dns",
					"abs_path": "node:dns",
					"lineno": 111,
					"colno": 26,
					"in_app": false
				}]
			},
			"mechanism": {
				"type": "generic",
				"handled": true
			}
		}]
	},
	"fingerprint": ["{{ default }}"],
	"grouping_config": {
		"enhancements": "eJybzDRxY3J-bm5-npWRgaGlroGxrpHxBABcYgcZ",
		"id": "newstyle:2019-10-29"
	},
	"hashes": ["cb142f47bc36a31540a7101553ebbcd3"],
	"ingest_path": [{
		"version": "22.11.0",
		"public_key": "[REDACTED]"
	}],
	"key_id": "2173646",
	"level": "error",
	"location": "node:dns",
	"logger": "",
	"metadata": {
		"display_title_with_tree_label": false,
		"filename": "node:dns",
		"function": "GetAddrInfoReqWrap.onlookup [as oncomplete]",
		"type": "Error",
		"value": "getaddrinfo ENOTFOUND posthog"
	},
	"nodestore_insert": 1670007802.112358,
	"received": 1670007798.438835,
	"sdk": {
		"name": "sentry.javascript.node",
		"version": "7.10.0",
		"integrations": ["InboundFilters", "FunctionToString", "ContextLines", "Console", "Http", "OnUncaughtException", "OnUnhandledRejection", "LinkedErrors", "Postgres"],
		"packages": [{
			"name": "npm:@sentry/node",
			"version": "7.10.0"
		}]
	},
	"timestamp": 1670007798.393,
	"title": "Error: getaddrinfo ENOTFOUND posthog",
	"type": "error",
	"version": "7"
}
```

**Comment by benjackwhite** [comment]
I would highly recommend joining our [posthog-users slack](https://posthog.com/slack) for the #community-support around self-hosted instances. These things can be hard to debug as everyone deploys it differently so there you can get the most real-time feedback and context sharing.

@PostHog/team-infra any ideas?

**Comment by joetrollo** [comment]
[This Slack thread](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1671482418043139) gave me a clue as to where this unexpected DNS lookup is coming from, and it's indeed the PostgreSQL user name. I've also identified that [this commit](https://github.com/PostHog/posthog/commit/f5b33390bd4068c98ac497a7444b35096c252e75) changed how the database connection information is given to `pg`, and landed between v1.40 and v1.41 which is right when my deployment broke. Changing the PostgreSQL user name to something different doesn't fix the issue like it did for the person in Slack—instead the error just changes to `getaddrinfo ENOTFOUND new_user_name`—and I don't believe the issue was actually a naming conflict as mentioned in that thread.

Rather, this is an issue with building the `DATABASE_URL`. Specifically, my issue was triggered by a `?` in the password. When PostHog constructs the `DATABASE_URL` from individual parts, the result is something like `postgres://username:pass?word@host:5432/database`, but when this is [given to the `Pool`](https://github.com/PostHog/posthog/blob/master/plugin-server/src/utils/utils.ts#L381) and ultimately `url.parse`, the result is:
```
Url {
  protocol: 'postgres:',
  slashes: true,
  auth: null,
  host: 'username',
  port: null,
  hostname: 'username',
  hash: null,
  search: '?word@host:5432/database',
  query: 'word@host:5432/database',
  pathname: '/:pass',
  path: '/:pass?word@host:5432/database',
  href: 'postgres://username/:pass?word@host:5432/database'
}
```

The plugin server should be URL-encoding the username and password before combining them into a URL.
